graphql-server/pkg/models/auth.go

```diff
@@ -1,15 +1,67 @@
 package models
 
 import (
+	"encoding/base64"
 	"fmt"
 	"time"
 
 	abnfparser "github.com/fkgi/abnf"
 	"github.com/oursky/likedao/pkg/abnf"
+	"github.com/tendermint/tendermint/crypto/secp256k1"
 )
 
 const DateLayout = "2006-01-02T15:04:05Z"
 
+type Fee struct {
+	Amount []string `json:"amount"`
+	Gas    string   `json:"gas"`
+}
+
+type MessageSignData struct {
+	Data   string `json:"data"` // Base64
+	Signer string `json:"signer"`
+}
+
+type Message struct {
+	Type  string          `json:"type"`
+	Value MessageSignData `json:"value"`
+}
+type PubKey struct {
+	Type  string `json:"type"`
+	Value string `json:"value"`
+}
+
+func (p *PubKey) ToSecp256k1() (*secp256k1.PubKey, error) {
```

Comment:
do we need to support other key types? https://docs.cosmos.network/master/basics/accounts.html#keys-accounts-addresses-and-signatures

Reply:
i think the community will be moving towards ed25519 in the future

Reply:
i see in the tendermint crypto package there's ed25519, secp256k1 and sr25519, will support those